activemq-client/src/main/java/org/apache/activemq/ActiveMQProducer.java

```diff
@@ -0,0 +1,547 @@
+            return this;
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean getDisableMessageID() {
+        try {
+            return activemqMessageProducer.getDisableMessageID();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setDisableMessageTimestamp(boolean value) {
+        try {
+            activemqMessageProducer.setDisableMessageTimestamp(value);
+            return this;
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean getDisableMessageTimestamp() {
+        try {
+            return activemqMessageProducer.getDisableMessageTimestamp();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setDeliveryMode(int deliveryMode) {
+        this.deliveryMode = deliveryMode;
+        return this;
+    }
+
+    @Override
+    public int getDeliveryMode() {
+        if(deliveryMode != null) {
+            return deliveryMode;
+        }
+
+        try {
+            return this.activemqMessageProducer.getDeliveryMode();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setPriority(int priority) {
+        this.priority = priority;
+        return this;
+    }
+
+    @Override
+    public int getPriority() {
+        if(priority != null) {
+            return priority;
+        }
+
+        try {
+            return this.activemqMessageProducer.getPriority();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setTimeToLive(long timeToLive) {
+        this.timeToLive = timeToLive;
+        return this;
+    }
+
+    @Override
+    public long getTimeToLive() {
+        if(timeToLive != null) {
+            return timeToLive;
+        }
+
+        try {
+            return this.activemqMessageProducer.getTimeToLive();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setDeliveryDelay(long deliveryDelay) {
+        this.deliveryDelay = deliveryDelay;
+        return this;
+    }
+
+    @Override
+    public long getDeliveryDelay() {
+        if(deliveryDelay != null) {
+            return deliveryDelay;
+        }
+
+        try {
+            return this.activemqMessageProducer.getDeliveryDelay();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
```

Comment:
Seems odd it would ever ask the shared _underlying_ MessageProducer what the delivery delay is. Its value shouldnt really change and would never seem to be relevant to _this_ JMSProducer.

Reply:
Ah, so actually this isnt really implemented, and the deliveryDelay value isnt used outside the get/set methods. Unclear why those are implemented yet then instead of throwing like so many others.

Reply:
Sure, soo per the second comment, why are these particular methods implemented at all then? Unlike the Message setter/getter that are required to support _other_ providers, these methods do actually imply Deliovery Delivery support is there when it is not. They should be throwing like all the other unimplemented bits.

Reply:
This has been changed